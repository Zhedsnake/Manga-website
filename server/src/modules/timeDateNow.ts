

export function timeDateNow(): string {
    const now = new Date();
    const hours = now.getHours();

    // Для docker
    const myHours = hours + 3;

    const stringMyHours = myHours.toString().padStart(2, '0');
    const minutes = now.getMinutes().toString().padStart(2, '0');
    const day = now.getDate();
    const monthNames = [
        "Январь", "Февраль", "Март", "Апрель", "Май", "Июнь",
        "Июль", "Август", "Сентябрь", "Октябрь", "Ноябрь", "Декабрь"
    ];
    const month = monthNames[now.getMonth()];
    const year = now.getFullYear();

    return ` ч:м ${stringMyHours}:${minutes}, день ${day}, месяц ${month}, год ${year}`;
}