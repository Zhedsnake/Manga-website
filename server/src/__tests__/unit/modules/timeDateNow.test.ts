import { timeDateNow } from '../../../modules/timeDateNow';
import MockDate from 'mockdate';

describe('timeDateNow', () => {

    beforeAll(() => {
        MockDate.set(new Date(2024, 8, 5, 13, 15))
    })

    afterAll(() => {
        MockDate.reset()
    })

    afterEach(() => {
        jest.restoreAllMocks();
    })

    test('Date должен быть вызван только один раз', () => {
        const spyDate = jest.spyOn(global, 'Date');
        timeDateNow();
        expect(spyDate).toHaveBeenCalledTimes(1);
    });

    test('Date методы должны быть вызваны один раз', () => {
        const spyGetHours = jest.spyOn(Date.prototype, 'getHours');
        const spyGetMinutes = jest.spyOn(Date.prototype, 'getMinutes');
        const spyGetDate = jest.spyOn(Date.prototype, 'getDate');
        const spyGetMonth = jest.spyOn(Date.prototype, 'getMonth');
        const spyGetFullYear = jest.spyOn(Date.prototype, 'getFullYear');

        timeDateNow();

        expect(spyGetHours).toHaveBeenCalledTimes(1);
        expect(spyGetMinutes).toHaveBeenCalledTimes(1);
        expect(spyGetDate).toHaveBeenCalledTimes(1);
        expect(spyGetMonth).toHaveBeenCalledTimes(1);
        expect(spyGetFullYear).toHaveBeenCalledTimes(1);
    });

    test('должна быть возвращена правильно отформатированная строка времени и даты', () => {
        expect(timeDateNow()).toBe(' ч:м 16:15, день 5, месяц Сентябрь, год 2024');
        expect(timeDateNow()).toMatchSnapshot();
    });

});
