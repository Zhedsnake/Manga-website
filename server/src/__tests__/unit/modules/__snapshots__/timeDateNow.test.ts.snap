// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`timeDateNow должна быть возвращена правильно отформатированная строка времени и даты 1`] = `" ч:м 16:15, день 5, месяц Сентябрь, год 2024"`;
